File: backend/Restaurant.Domain/Entities/Booking.cs
namespace Restaurant.Domain.Entities;

public class Booking
{
    public int Id { get; set; }
    public int TableId { get; set; }
    public int? ClientId { get; set; }
    public string ClientName { get; set; } = string.Empty;
    public string ClientPhone { get; set; } = string.Empty;
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public string? Comment { get; set; }
    public string Status { get; set; } = "активно"; // активно, завершено, отменено, конвертировано
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Table? Table { get; set; }
    public Client? Client { get; set; }
    public ICollection<Order>? Orders { get; set; }
}

File: Prog_korp_seti-master/types/booking.ts
/**
 * Статус бронирования
 */
export type BookingStatus = 'Active' | 'Cancelled' | 'Completed';

/**
 * Бронирование
 */
export interface Booking {
  id: number;
  clientName: string;
  clientPhone?: string; // Полный номер для админа
  phoneMasked: string; // Маскированный телефон для безопасности
  phoneLastFour: string; // Последние 4 цифры для поиска
  start: string; // ISO 8601
  end: string; // ISO 8601
  comment?: string;
  guestCount?: number; // Количество гостей
  tableId: number;
  table?: {
    id: number;
    location: string;
    seats: number;
  };
  status: BookingStatus;
  createdAt: string;
  updatedAt?: string;
}

/**
 * Создание бронирования
 * Соответствует CreateBookingDto на бэкенде
 */
export interface BookingCreate {
  TableId: number; // PascalCase для совместимости с .NET
  ClientName: string; // PascalCase для совместимости с .NET
  ClientPhone: string; // Полный номер телефона, PascalCase
  StartTime: string; // ISO 8601, PascalCase
  EndTime: string; // ISO 8601, PascalCase
  Comment?: string; // PascalCase
}

/**
 * Поиск бронирования
 */
export interface BookingSearchParams {
  name: string;
  phoneSuffix: string; // Последние 4 цифры
}

/**
 * Параметры для поиска доступных столов
 */
export interface TableAvailabilityQuery {
  start: string; // ISO 8601
  end: string; // ISO 8601
  seats: number;
  location?: string;
}
